### DISCUSSION TITLE
Documentation for how the code works

### DISCUSSION THREAD

**Discussion by jibin23** [noDRM/DeDRM_tools#352] [General]
I have been trying to read the source code to understand how the DeDRM process works, however I am getting lost.
Can one of the contributors write an overview of what happens in the code, and which .py scripts are run first in any use case? We can start with Amazon .azw3 DRM.

This will enable me to contribute, if I can, to this repo

**Comment by ElleKayEm** [comment]
If you're not familiar with calibre plugins in general, you could start here: https://manual.calibre-ebook.com/creating_plugins.html#pluginstutorial

**Reply by jibin23** [reply]
Thanks for this. But my comment was focused on the DeDRM tech, not the Calibre Plugin wrapper around it.

**Reply by noDRM** [reply]
How exactly the code works depends on the DRM in question. The __init__.py file contains the code executed by Calibre upon encountering a DRM'ed file, and in there are the checks to find out which DRM is present in the file. In there, the plugin then calls functions for the detected DRM that are in other files, like ion.py for Amazon, ineptepub.py or ineptpdf.py for Adobe, or (in the past) lcpdedrm.py for the Readium DRM.

Then, most of the code for configuration and Calibre menus is in config.py and prefs.py.
